engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ProcessingStateMachine.java

```diff
@@ -446,15 +446,21 @@ public boolean isMakingProgress() {
     return !onErrorHandlingLoop;
   }
 
-  public void startProcessing(final long lastProcessedPosition) {
+  public void startProcessing(final LastProcessingPositions lastProcessingPositions) {
     // Replay ends at the end of the log and returns the lastSourceEventPosition
     // which is equal to the last processed position
     // we need to seek to the next record after that position where the processing should start
     // Be aware on processing we ignore events, so we will process the next command
+    final var lastProcessedPosition = lastProcessingPositions.getLastProcessedPosition();
     logStreamReader.seekToNextEvent(lastProcessedPosition);
     if (lastSuccessfulProcessedEventPosition == StreamProcessor.UNSET_POSITION) {
       lastSuccessfulProcessedEventPosition = lastProcessedPosition;
     }
+
+    if (lastWrittenEventPosition == StreamProcessor.UNSET_POSITION) {
+      lastWrittenEventPosition = lastProcessingPositions.getLastWrittenPosition();
```

Comment:
🔧  While we're at it, let's rename this field to what it actually is: `lastWrittenPosition` (it does not necessary only relate to events).

Reply:
true